storage/innobase/os/os0file.cc

```diff
@@ -1095,13 +1095,11 @@ static ATTRIBUTE_COLD void os_file_log_buffered()
   log_sys.log_maybe_unbuffered= false;
   log_sys.log_buffered= true;
 }
-# endif
 
 /** @return whether the log file may work with unbuffered I/O. */
 static ATTRIBUTE_COLD bool os_file_log_maybe_unbuffered(const struct stat &st)
 {
   MSAN_STAT_WORKAROUND(&st);
-# ifdef __linux__
   char b[20 + sizeof "/sys/dev/block/" ":" "/../queue/physical_block_size"];
   if (snprintf(b, sizeof b, "/sys/dev/block/%u:%u/queue/physical_block_size",
                major(st.st_dev), minor(st.st_dev)) >=
```

Comment:
Do all operating systems that support `O_DIRECT` provide a Linux-like interface for determining the physical block size of a storage device?

Reply:
> It would be good to mention why https://github.com/MariaDB/server/commit/5369df741b3a87d7b262aafe95438b33ed124295 was not applied to 10.6 or 10.5.

This looks more likely to be the problem than the merge itself, as it is natural to keep a file deleted in a conflict of merging from a lower version to a higher version, if the lower version modifies the file and the file is deleted in a higher version.

Reply:
Even if a cherry-pick of 5369df741b3a87d7b262aafe95438b33ed124295 had been included in the merge, `git merge` would have flagged the conflict as "deleted by us". Such conflicts can be understood by reading the output of `git log HEAD..MERGE_HEAD filename` or `git log MERGE_HEAD..HEAD filename` and resolved by studying the identified commits in more detail.